gapic-common/lib/gapic/common/retry_policy.rb

```diff
@@ -169,15 +182,9 @@ def increment_delay!
       end
 
       # @private
-      # @return [Numeric] The deadline for timeout-based policies based policies.
+      # @return [Numeric] The deadline for timeout-based policies.
       def deadline
-        @deadline ||= Process.clock_gettime(Process::CLOCK_MONOTONIC) + timeout
-      end
-
-      # @private
-      # @return [Boolean] Whether this policy should be retried based on the deadline.
-      def retry_with_deadline?
-        deadline > Process.clock_gettime(Process::CLOCK_MONOTONIC)
+        @deadline ||= update_deadline!
```

Comment:
I think this would be less confusing if you used `||` rather than `||=`. Since `update_deadline!` already sets `@deadline`. I had to think a bit to convince myself the logic is correct, and the assignment threw me off.

Reply:
The logic has changed to be not based on process clock. PTAL again.